src/Catalog/CatalogTypeConverter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public class CatalogTypeConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(IEnumerable<string>);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            object type = null;
+            if (reader.TokenType == JsonToken.StartArray)
+            {
+                // If the type is stored in an array, use the first value and then discard the rest.
+                do
+                {
+                    reader.Read();
+                    if (reader.TokenType == JsonToken.String)
+                    {
+                        type = reader.Value;
```

Comment:
>Value [](start = 38, length = 5)

Is there a specific reason why `type` is an object and you cast `type` to string at the very end? If not, you could simplify the code a little by making `type` a string from the get-go and use `reader.ReadAsString` instead here.

Reply:
I think this is better...hard to explain with just text but using ReadAsString would require me to move around the rest of the flow and I think this is simpler.